### PR TITLE
Completer

### PR DESCRIPTION
**Pull request by coreyadkins**

### CODE REVIEW

File: practice/completer/site.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+var completerPrototype = {
+  addCompletion: function(comp) {
+    this.completions.push(comp);
+  },
+  removeCompletion: function(str) {
+    var index = this.completions.indexOf(str);
+    this.completions.splice(index, 1);
```
Comment (schmamps) [inline practice/completer/site.js]: This deletes the last item of the array if str does not exist.

File: practice/completer/site.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+var completerPrototype = {
+  addCompletion: function(comp) {
+    this.completions.push(comp);
```
Comment (schmamps) [inline practice/completer/site.js]: What about duplicate values?

File: practice/completer/site.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  removeCompletion: function(str) {
+    var index = this.completions.indexOf(str);
+    this.completions.splice(index, 1);
+  },
+  complete: function(prefix) {
+    var completionsArray = [];
+    completionsArray.push(_.filter(this.completions, function(comp) {
+      return _.startsWith(comp, prefix);
+    }));
+    return completionsArray;
```
Comment (schmamps) [inline practice/completer/site.js:31]: This looks like the long way around using _.filter. Off the top of my head it looks like:

``` js
var completionsArray = _.filter(this.completions, function (comp) { 
    return _.startsWith(comp, prefix); 
});
```

File: practice/completer/site.js (hunk 2 above)
Comment (coreyadkins) [inline practice/completer/site.js]: Probably a good thing to think about, although I don't think it's totally necessary to solve the basic problem. Simple enough with `_.includes` that I might as well use it, thanks!

File: practice/completer/site.js (hunk 1 above)
Comment (coreyadkins) [inline practice/completer/site.js]: Thanks, good catch.

File: practice/completer/site.js (hunk 2 above)
Comment (schmamps) [inline practice/completer/site.js]: Apparently the simple way in lodash is:

``` js
this.completions = _.union(this.completions, [comp]);
```

Comment (schmamps) [inline practice/completer/site.js]: It's good enough that I added it to my own version :)

``` js
this.completions = _.union(this.completions, [str]).sort();
```

File: practice/completer/site.js (hunk 1 above)
Comment (schmamps) [inline practice/completer/site.js]: Stephen has an even better solution: _.pull()

File: practice/completer/site.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+var completerPrototype = {
+  /**
+   * Adds a word to the completion list to be stored as a suggestion.
+   */
+  addCompletion: function(comp) {
+    if (_.includes(this.completions, comp)) {
+      null; //Is this the best way to tell JS to do nothing?
+    } else {
+      this.completions.push(comp);
+    }
```
Comment (davidselassie) [inline practice/completer/site.js]: If you only want the "else", you can negate the bool expression:

``` js
if (!_.includes(this.completions, comp)) {
...
}
```

Comment (coreyadkins) [inline practice/completer/site.js]: Sometimes the answer is too obvious to see. Thanks!

**Pull request merged**